forte/data/data_store.py

```diff
@@ -318,18 +320,32 @@ def _get_type_parent(self, type_name: str) -> str:
         """
         return self._get_type_info(type_name)["parent_class"]
 
-    def _num_attributes_for_type(self, type_name: str) -> int:
-        """Get the length of the attribute dict of an entry type.
+    def _default_attributes_for_type(self, type_name: str) -> List:
+        """Get a list of attributes of an entry type with their default values.
+        If an attribute is annotated with `FList` or `List`, then the default
+        value is an empty list `[]`. When an attribute is annotated with `FDict`
+        or `Dict` then the default value will be an empty dictionary `{}`. For
+        all other cases (including primitive types, Union, NoneType, etc.) the
+        default value will be set to `None`.
+
         Args:
             type_name (str): The fully qualified type name of the new entry.
         Returns:
-            attr_dict (dict): The attributes-to-index dict of an entry.
+            attr_dict (list): A list of attributes with default values.
         """
-        return len(self._get_type_attribute_dict(type_name))
+        attr_dict: Dict = self._get_type_attribute_dict(type_name)
+        attr_fields: Dict = self._get_entry_attributes_by_class(type_name)
+        attr_list: List = [None] * len(attr_dict)
+        for attr_name, attr_id in attr_dict.items():
+            attr_class = get_origin(attr_fields[attr_name].type)
```

Comment:
This is now inspect on the fly, but it is probably better to keep a record of this. Maybe mark this as a todo?

Reply:
or otherwise a simple approach is to cache this in the class